packages/bundler-plugin-core/src/plugins/react-annotate-plugin.js

```diff
@@ -0,0 +1,480 @@
+      FunctionDeclaration(path, state) {
+        if (!path.node.id || !path.node.id.name) return;
+        if (isKnownIncompatiblePluginFromState(state)) return;
+        functionBodyPushAttributes(
+          state.opts[annotateFragmentsOptionName] === true,
+          t,
+          path,
+          path.node.id.name,
+          sourceFileNameFromState(state),
+          attributeNamesFromState(state),
+          this.excludedComponents
+        );
+      },
+      ArrowFunctionExpression(path, state) {
+        if (!path.parent.id || !path.parent.id.name) return;
+        if (isKnownIncompatiblePluginFromState(state)) return;
+        functionBodyPushAttributes(
+          state.opts[annotateFragmentsOptionName] === true,
+          t,
+          path,
+          path.parent.id.name,
+          sourceFileNameFromState(state),
+          attributeNamesFromState(state),
+          this.excludedComponents
+        );
+      },
+      ClassDeclaration(path, state) {
+        const name = path.get("id");
+        const properties = path.get("body").get("body");
+        const render = properties.find((prop) => {
+          return prop.isClassMethod() && prop.get("key").isIdentifier({ name: "render" });
+        });
+
+        if (!render || !render.traverse) return;
+        if (isKnownIncompatiblePluginFromState(state)) return;
+
+        const excludedComponents = this.excludedComponents;
+
+        render.traverse({
+          ReturnStatement(returnStatement) {
+            const arg = returnStatement.get("argument");
+
+            if (!arg.isJSXElement() && !arg.isJSXFragment()) return;
+            processJSX(
+              state.opts[annotateFragmentsOptionName] === true,
+              t,
+              arg,
+              name.node && name.node.name,
+              sourceFileNameFromState(state),
+              attributeNamesFromState(state),
+              excludedComponents
+            );
+          },
+        });
+      },
+    },
+  };
+}
+
+function fullSourceFileNameFromState(state) {
+  const name = state.file.opts.parserOpts.sourceFileName;
+  if (typeof name !== "string") {
+    return undefined;
+  }
+  return name;
+}
+
+function sourceFileNameFromState(state) {
+  const name = fullSourceFileNameFromState(state);
+  if (name === undefined) {
+    return undefined;
+  }
+
+  if (name.indexOf("/") !== -1) {
+    return name.split("/").pop();
+  } else if (name.indexOf("\\") !== -1) {
+    return name.split("\\").pop();
+  } else {
+    return name;
+  }
+}
+
+function isKnownIncompatiblePluginFromState(state) {
+  const fullSourceFileName = fullSourceFileNameFromState(state);
+  if (fullSourceFileName == undefined) {
+    return false;
+  }
+
+  for (let i = 0; i < knownIncompatiblePlugins.length; i += 1) {
+    let pluginName = knownIncompatiblePlugins[i];
+    if (
+      fullSourceFileName.includes("/node_modules/" + pluginName + "/") ||
+      fullSourceFileName.includes("\\node_modules\\" + pluginName + "\\")
```

Comment:
l: suprised this is not turned on as eslint rule, but we should use template syntax

```suggestion
      fullSourceFileName.includes(`/node_modules/${pluginName}/`) ||
      fullSourceFileName.includes(`\\node_modules\\${pluginName}\\`)
```